### DISCUSSION TITLE
Timeout for blocking_send

### DISCUSSION THREAD

**Discussion by prkadalbcisco** [tokio-rs/tokio#5421] [Q&A] [answered]
Hello everyone,

I use `span_blocking` to execute synchronous code and an `mpsc::channel` to get data back from it.

If the synchronous code has called a `blocking_send` on the channel and is waiting there, and the async code wants to exit, will closing the channel suffice? Will the `blocking_send` immediately error out and release the synchronous code to perform its cleanup and exit? If not, is it possible to set some kind of a timeout on the `blocking_send`?

Edit: 
The async code on the receiving end of the channel is a rocket endpoint; so if the browser/API caller hangs up abruptly, it may be left abandoned and the blocking_send will wait indefinitely

**Comment by Darksonn** [comment, marked as answer]
The `blocking_send` method will immediately return an error if the receiver is destroyed.

You can get a timeout by doing this:
```
handle.block_on(tokio::time::timeout(duration, sender.send(msg)));
```
where `handle` is a `tokio::runtime::Handle` you got from your runtime.

**Reply by prkadalbcisco** [reply]
Thank you
